docs/topics/http/urls.txt

```diff
@@ -368,6 +368,21 @@ the following example is valid::
 In the above example, the captured ``"username"`` variable is passed to the
 included URLconf, as expected.
 
+Nested captured parameters, for example ``(?P<outer>(?P<inner>\w+))`` or
+``((\d+)/)?``, are discarded by Django when urls are reversed, using only the
+outer parameters to find a reverse for a pattern. If you want to reverse such
+patterns, only pass the outer argument or named argument.
+
+For cases such as making a part of the pattern optional, use non-capturing
+arguments -- for example when a non-query page-specifier is optional::
+
+    # In urls.py
+    from django.conf.urls import url
+
+    urlpatterns = [
+        url(r'^(?P<username>\w+)/(?:page-(?P<page>\d+)/)?$', blog_entry),
```

Comment:
We could say something like "Nested captured parameters, while not recommended due to their complexity, ..." and omit everything after "For cases such as".